src/USER-MISC/compute_pressure_cylinder.cpp

```diff
@@ -0,0 +1,493 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "math.h"
+#include "string.h"
+#include "stdlib.h"
+#include "compute_pressure_cylinder.h"
+#include "atom.h"
+#include "update.h"
+#include "force.h"
+#include "pair.h"
+#include "neighbor.h"
+#include "neigh_request.h"
+#include "neigh_list.h"
+#include "group.h"
+#include "memory.h"
+#include "error.h"
+
+using namespace LAMMPS_NS;
+
+static const char cite_compute_pressure_cylinder[] =
+  "compute pressure/cylinder:\n\n"
+  "@Article{Addington,\n"
+  " author = {C. K. Addington, Y. Long, K. E. Gubbins},\n"
+  " title = {The pressure in interfaces having cylindrical geometry},\n"
+  " journal = {J.~Chem.~Phys.},\n"
+  " year =    2018,\n"
+  " volume =  149,\n"
+  " pages =   {084109}\n"
+  "}\n\n";
+
+if (lmp->citeme) lmp->citeme->add(cite_compute_pressure_cylinder);
+  
+/*+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  Calculate the configurational components of the pressure tensor in 
+  cylindrical geometry, according to the formulation of Addington et al. (2018)
+  +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ */
+
+ComputePressureCyl::ComputePressureCyl(LAMMPS *lmp, int narg, char **arg) :
+  Compute(lmp, narg, arg)
+{
+  if (narg != 7) error->all(FLERR,"Illegal compute pressure/cylinder command");
+
+  zlo=force->numeric(FLERR,arg[3]);
+  zhi=force->numeric(FLERR,arg[4]);
+  Rmax=force->numeric(FLERR,arg[5]);
+  bin_width=force->numeric(FLERR,arg[6]);
+
+  nbins=(int)(Rmax/bin_width);
+
+  nzbins=(int)((zhi-zlo)/bin_width);
+
+  array_flag=1;
+  vector_flag=0;
+  extarray=0;
+  size_array_cols = 5;  // r, number density, Pr, Pphi, Pz
+  size_array_rows = nbins;
+
+  Pr_temp = new double[nbins];
```

Comment:
Please initialize all these pointers (like Pr_temp) to NULL in the constructor init. See for example vptr, and id_temp in compute_pressure.cpp (line 40).
This is required for the library interface when configured to throw an exception on errors instead of exiting.

Reply:
Will do! Thanks for looking this over.